### PR TITLE
Migration Support

### PR DESCRIPTION
**Pull request by woolfg**

### CODE REVIEW

File: db_schema/schema.sql [hunk 1]
```diff
@@ -1,3 +1,19 @@
+-- Open Podcast Analytics database schema
+
+-- migration table to track applied migrations
+CREATE TABLE IF NOT EXISTS migrations (
+  migration_id INTEGER NOT NULL,
+  migration_name VARCHAR(64) NOT NULL,
+  migration_date TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (migration_id)
+);
```
Comment (mre) [inline db_schema/schema.sql:9]: I'm confused why this is here if it's also in `migrations/1.sql1`. Can we add an explanation why we need the duplication or was that a mistake?

File: db_schema/migrations/1.sql
Comment (mre) [inline db_schema/migrations/1.sql:1]: What do you think about adding a short description for every migration? E.g. `1_init_migrations_table.sql`.
I found that it helps quickly when scanning through the directory of migrations without looking into each file.

File: src/db/DBInitializer.ts [hunk 2]
```diff
@@ -72,18 +80,72 @@ class DBInitializer {
...
         const tablesExist = await this.checkTablesExist()
         if (!tablesExist) {
             console.log(
                 "MySQL tables don't exist in the specified db, creating them..."
             )
             await this.runQueries(this.schemaData)
             console.log('tables created')
+            // if there is already a schema, check if migrations are needed
+        } else {
+            await this.runMigrations()
```
Comment (mre) [inline src/db/DBInitializer.ts:144]: Shouldn't we unconditionally run the migrations?
Say a table didn't exist, so we create it, but then we don't run the migrations on that table and there is a migration for the fields for that table, or we added an index. Then the migrations might not run, right?

File: src/db/DBInitializer.ts [hunk 3]
```diff
@@ -72,18 +80,72 @@ class DBInitializer {
...
         }
+        // queries that are always executed, e.g. to replace all views
         if (this.sqlToRun) {
-            console.log('Running sql statements ...')
+            console.log(`Running sql statements ...`)
```
Comment (mre) [inline src/db/DBInitializer.ts:148]: Does the formatter always change to backticks?

Comment (woolfg) [reply src/db/DBInitializer.ts:148]: nope, used a var but removed it again

File: src/db/DBInitializer.ts (hunk 2 above)
Comment (woolfg) [reply src/db/DBInitializer.ts:144]: the schema.sql always contains the up to date schema. therefore, it is not needed to run any migrations if the schema is deployed on a fresh installation

File: db_schema/schema.sql (hunk 1 above)
Comment (woolfg) [reply db_schema/schema.sql:9]: good idea

File: db_schema/migrations/1.sql
Comment (woolfg) [reply db_schema/migrations/1.sql:1]: good idea

File: db_schema/schema.sql (hunk 1 above)
Comment (woolfg) [reply db_schema/schema.sql:9]: nope, it is on purpose. will make it more clear.

Comment (woolfg) [reply db_schema/schema.sql:15]: see docs in schema.sql, is it clearer now?

Comment (mre) [reply db_schema/schema.sql:15]: ah got it. thanks

**Review by mre — APPROVED** [review]
lgtm

**Pull request merged**